### PR TITLE
Add name parameter to Measurement class

### LINKED ISSUE #1736: Add name parameter to Measurement class
The `Measurement` class has the attribute `name`, which is currently hardcoded to an empty string ([here](https://github.com/QCoDeS/Qcodes/blob/master/qcodes/dataset/measurements.py#L686)). Because of this, doing `with meas.run() as datasaver: ...` will always give the dataset the name `"results"` ([here](https://github.com/QCoDeS/Qcodes/blob/master/qcodes/dataset/measurements.py#L582)). Would it be possible to add a `name` argument to the Measurement constructor? So that the `__repr__` of a dataset could be slightly more informative.

Or does this mess up something with the database? As in https://github.com/QCoDeS/Qcodes/issues/898

### PR DESCRIPTION
**Pull request by mgunyho**
Fixes #1736.

I added the name as the first kwarg to be consistent with other classes, and also a deprecation warning if the first parameter is instead an Experiment.

### CODE REVIEW

**Comment by mgunyho** [comment]
I made the changes you requested. Sorry about not adding a test previously, I didn't look in the subfolders. I hope the test is not too simple.

File: qcodes/dataset/measurements.py [hunk 1]
```diff
@@ -668,36 +667,27 @@ class Measurement:
...
+            produced by the measurement. If not given, a default value of
+            'results' is used for the dataset.
     """
 
-    def __init__(self, name: Optional[str] = '',
-                 exp: Optional[Experiment] = None,
-                 station: Optional[qc.Station] = None) -> None:
+    def __init__(self, exp: Optional[Experiment] = None,
+                 station: Optional[qc.Station] = None,
+                 name: Optional[str] = '') -> None:
```
[the remark this answers is not in the document]
Comment (mgunyho) [reply qcodes/dataset/measurements.py]: True, the default value is the empty string, not None.

File: qcodes/tests/dataset/test_measurement_context_manager.py [hunk 2]
```diff
@@ -416,6 +416,23 @@ def test_measurement_name(experiment, DAC, DMM):
         assert datasaver.dataset.table_name == expected_name
 
 
+def test_measurement_name_param(experiment, DAC, DMM):
+    fmt = experiment.format_string
+    exp_id = experiment.exp_id
+
+    name = 'yolo'
+
+    meas = Measurement(name=name)
```
Comment (astafan8) [inline qcodes/tests/dataset/test_measurement_context_manager.py]:
```suggestion
    meas = Measurement(name=name, exp=experiment)
```

File: qcodes/tests/dataset/test_measurement_context_manager.py [hunk 3]
```diff
@@ -416,6 +416,23 @@ def test_measurement_name(experiment, DAC, DMM):
...
+
+    meas.register_parameter(DAC.ch1)
+    meas.register_parameter(DMM.v1, setpoints=[DAC.ch1])
+
+    with meas.run() as datasaver:
+        run_id = datasaver.run_id
+        expected_name = fmt.format(name, exp_id, run_id)
+        assert datasaver.dataset.table_name == expected_name
```
Comment (astafan8) [inline qcodes/tests/dataset/test_measurement_context_manager.py]: also assert name of the dataset
```suggestion
        assert datasaver.dataset.table_name == expected_name
        assert datasaver.dataset.name == name
```

File: qcodes/tests/dataset/test_measurement_context_manager.py [hunk 4]
```diff
@@ -416,6 +416,23 @@ def test_measurement_name(experiment, DAC, DMM):
         assert datasaver.dataset.table_name == expected_name
 
 
+def test_measurement_name_param(experiment, DAC, DMM):
```
Comment (astafan8) [inline qcodes/tests/dataset/test_measurement_context_manager.py]: could you also add a test where name argument is not passes and assert that the name of the dataset becomes "results"?

**Pull request merged**